### PR TITLE
Bug 4066 Closing Stash Form with Escape button while FileViewer(CodeDiff) is focused

### LINKED ISSUE #4066: Stash dialog can't be closed by Escape key
**Do you want to request a *feature* or report a *bug*?**
Bug.

**What is the current behavior?**
Stash dialog refuses to be closed by pressing Escape key if diff panel is focused.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
* open repository
* view stash
* click somewhere in code diff viewer to focus it (caret is starting to blink in it)
* try to press Escape key - nothing is happened
* click on changed file list on the left side, press Escape - Stash dialog is closed

**What is the expected behavior?**
Guess, dialog should be closed if diff viewer is focused unless this is intended behavior.

**Environment you encounter the issue:**

- GitExtensions version: 2.50.02
- GIT version: git version 2.14.2.windows.2
- OS version: Windows 7 SP1 x64

**Did this work in previous version of GitExtensions (which)?**
I think it worked well in previous versions, but I can't be 100% sure.

### PR DESCRIPTION
**Pull request by mhamdy88**
Fixes #4066 

Changes proposed in this pull request:
- bubble up KeyUp event at FileViewer control
- Add KeyUp event Handler to FileViewer control at StashForm to close the form while pressing Escape

### CODE REVIEW

**Comment by RussKie** [comment]
It looks like `FormStash` has received a new button. Please post a screenshot with the new look.

File: GitUI/CommandsDialogs/FormStash.cs [hunk 1]
```diff
@@ -378,5 +378,14 @@ private void toolStripButton_customMessage_EnabledChanged(object sender, EventAr
                 StashMessage.ReadOnly = false;
             }
         }
+
+        private void View_KeyUp(object sender, KeyEventArgs e)
+        {
+            // Close Stash form with escape button while pointer focus is in FileViewer(diff view)
+            if (e.KeyCode == Keys.Escape)
+            {
+                Cancel.PerformClick();
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormStash.cs]: Is there stopping us from calling the form's `Close()` and closing the form?

**Comment by mhamdy88** [comment]
Suppose to be at the back of other controls (send to back).. nothing to
show in new screebshot

Still should i post before/after screenshots?

**Comment by mhamdy88** [comment]
I dont think so, i was following the same pattern at FormCommit

File: GitUI/CommandsDialogs/FormStash.cs (hunk 1 above)
Comment (mhamdy88) [reply GitUI/CommandsDialogs/FormStash.cs]: I dont think so, i was following the same pattern at FormCommit

Comment (RussKie) [reply GitUI/CommandsDialogs/FormStash.cs]: Hmm... thanks, `FormCommit` will need to get looked at.
Let's not add a new button here, just call `Close()`, if there is nothing stopping us from doing it.

**Comment by RussKie** [comment]
Please resolve merge conflicts and squash

**Comment by RussKie** [comment]
Thank you

**Pull request merged**